### PR TITLE
feat(grpc): add grpc support for the chat service 

### PR DESCRIPTION
**Pull request by giovaz94**
Add support for GRPC calls inside the Chat Service. 
This PR includes also a better test support for interacting with cassandra db.

### CODE REVIEW

File: grpc/src/test/scala/ServiceHandlerTest.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import io.github.positionpal.proto.{Message, MessageResponse, RetrieveLastMessagesRequest}
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import scalapb.UnknownFieldSet
+
+class ServiceHandlerTest extends AnyWordSpec with BeforeAndAfterAll with Matchers with ScalaFutures:
+
+  val conf = ConfigFactory.parseString("""
```
Comment (v4l3rio) [inline grpc/src/test/scala/ServiceHandlerTest.scala:16]: you can put config in a separate file?

File: storage/build.gradle.kts [hunk 2]
```diff
@@ -12,47 +10,9 @@ dependencies {
...
-
-            if (!healthy) {
-                Thread.sleep(2000)
-                attempts++
-                println("Waiting for Cassandra to be healthy... Attempt ${attempts}/${maxAttempts}")
-            }
+dockerCompose {
 
+    
+}
```
Comment (tassiluca) [inline storage/build.gradle.kts:16]: If empty I think you can remove it
```suggestion
```

File: common/src/test/resources/db/cassandra_data/snapshot.tar.gz
Comment (tassiluca) [inline common/src/test/resources/db/cassandra_data/snapshot.tar.gz:1]: What is this?

**Review by tassiluca — APPROVED** [review]

File: common/src/test/resources/db/cassandra_data/snapshot.tar.gz
Comment (giovaz94) [reply common/src/test/resources/db/cassandra_data/snapshot.tar.gz:1]: Is a dump of the cassandra db. I use it for testing puropose

Comment (giovaz94) [reply common/src/test/resources/db/cassandra_data/snapshot.tar.gz:1]: Is possible to create a snapshot of the db from ./scripts/backup_cassandra.sh and restore from one of these using, inside a container, ./scripts/restore_cassandra.sh

**Pull request merged**